Solutions/Threat Intelligence/Analytic Rules/DomainEntity_CommonSecurityLog.yaml

```diff
@@ -20,64 +20,61 @@ triggerThreshold: 0
 tactics:
   - Impact
 query: |
-  let dt_lookBack = 1h;
-  let ioc_lookBack = 14d;
-  let IoCList = externaldata(TimeGenerated:datetime,IoC:string,IoC_Type:string,ExpirationDateTime:datetime,Description:string,Action:string, ConfidenceScore:real, ThreatType:string, Active:string,Type:string, TrafficLightProtocolLevel:string, ActivityGroupNames:string)[@"https://raw.githubusercontent.com/microsoft/mstic/master/RapidReleaseTI/Indicators.csv"] with(format="csv", ignoreFirstRecord=True);
-  //Create a list of TLDs in our threat feed for later validation of extracted domains
-  let list_tlds = ThreatIntelligenceIndicator
-  | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
-  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
-  | where Active == true
-  | where isnotempty(DomainName)
-  | extend DomainName = tolower(DomainName)
-  | extend parts = split(DomainName, '.')
-  | extend tld = parts[(array_length(parts)-1)]
-  | summarize count() by tostring(tld)
-  | summarize make_list(tld);
-  let Domain_Indicators = (union isfuzzy=true
-  (ThreatIntelligenceIndicator
-  | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
-  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
-  | where Active == true
-  // Picking up only IOC's that contain the entities we want
-  | where isnotempty(DomainName)
-  | extend TI_DomainEntity = DomainName
-  ),
-  (IoCList
-    | where IoC_Type =~ 'domainname'
-    | where ExpirationDateTime > now()
-    | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IoC
-    | where Active =~ 'True'
-    | extend TI_DomainEntity = IoC
-    | project-away  IoC_Type
-  )
-  );
+  // Define the time range for data retrieval
+  let dt_lookBack = 1h; // Look back 1 hour
+  let ioc_lookBack = 14d; // Look back 14 days
+  // Create a list of top-level domains (TLDs) from the threat feed data for later validation
+  let list_tlds =
+      ThreatIntelligenceIndicator
+      // Filter indicators based on the specified time range and active indicators
+      | where TimeGenerated >= ago(ioc_lookBack)
+      | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
+      | where Active == true and ExpirationDateTime > now()
+      | where isnotempty(DomainName)
+      // Convert domain names to lowercase for consistency
+      | extend DomainName = tolower(DomainName)
+      // Split domain names into parts and extract the TLD
+      | extend parts = split(DomainName, '.')
+      | extend tld = parts[(array_length(parts) - 1)]
+      // Count the occurrences of each TLD
+      | summarize count() by tostring(tld)
+      // Create a list of TLDs
+      | summarize make_list(tld);
+  // Retrieve threat intelligence indicators within the specified time range
+  let Domain_Indicators =
+      ThreatIntelligenceIndicator
+      | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
```

Comment:
This condition (and ExpirationDateTime > now()) should be present with Active == true.

Reply:
Instead of moving Active == true in line 46, move ExpirationDateTime > now() in line 48

Reply:
Moved the ExpirationDateTime > now() to line 48